File: problems-extend/1201-1300/1221_split-a-string-in-balanced-strings/solution.go
func balancedStringSplit(s string) int {
	// +1 for L, -1 for R: every return to zero is one more balanced piece,
	// and cutting at each is the finest valid split.
	balance, pieces := 0, 0
	for i := 0; i < len(s); i++ {
		if s[i] == 'L' {
			balance++
		} else {
			balance--
		}
		if balance == 0 {
			pieces++
		}
	}
	return pieces
}

File: problems-extend/1201-1300/1221_split-a-string-in-balanced-strings/solution.cpp
class Solution {
  public:
    int balancedStringSplit(string s) {
        // +1 for L, -1 for R: every return to zero is one more balanced
        // piece, and cutting at each is the finest valid split.
        int balance = 0, pieces = 0;
        for (char ch : s) {
            balance += ch == 'L' ? 1 : -1;
            if (balance == 0)
                ++pieces;
        }
        return pieces;
    }
};
